components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -616,6 +616,21 @@ impl<E: KvEngine> CoprocessorHost<E> {
         }
     }
 
+    pub fn pre_delete_range(&self, start_key: &[u8], end_key: &[u8]) {
+        let region = Region::default();
+        let mut ctx = ObserverContext::new(&region);
+        for observer in &self.registry.query_observers {
+            let observer = observer.observer.inner();
+            let mut request = Request::new();
+            request.set_cmd_type(CmdType::DeleteRange);
+            request.mut_delete_range().set_start_key(start_key.to_vec());
+            request.mut_delete_range().set_end_key(end_key.to_vec());
```

Comment:
To maintain consistency in the request, it's best not to include data keys (with a 'z' prefix) in the request. In this pull request, the caller (`GcRunnerCore`) is passing data keys, which is not recommended.